aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move

```diff
@@ -0,0 +1,48 @@
+/// This module provides an interface for aggregators (version 2).
+/// Only skeleton - for AggregagtorSnapshot - is provided at this time,
+/// to allow transition of usages.
+module aptos_framework::aggregator_v2 {
+    use std::string::String;
+
+    struct AggregatorSnapshot<Element> has store, drop {
+        value: Element,
+    }
+
+    public native fun create_snapshot<Element: copy + drop>(value: Element): AggregatorSnapshot<Element>;
+
+    public native fun copy_snapshot<Element: copy + drop>(snapshot: &AggregatorSnapshot<Element>): AggregatorSnapshot<Element>;
+
+    public native fun read_snapshot<Element>(snapshot: &AggregatorSnapshot<Element>): Element;
+
+    public native fun string_concat<Element>(before: String, snapshot: &AggregatorSnapshot<Element>, after: String): AggregatorSnapshot<String>;
```

Comment:
Why `string_concat`? I forgot what was the name we decided on but `format_snapshot` or something like that seems more appropriate? No strong preference :)

Reply:
I don't have a strong preference either. @igor-aptos actually used this function name in his original PR. So, I just the copied the names from there.

Reply:
we reserve format for more general formatting, here it's concat of snapshot with string prefix and suffix which is enough but not quite what you'd expect from formatting interfaces.